src/pages/ProductDetail/index.tsx

```diff
@@ -0,0 +1,81 @@
+import styled from '@emotion/styled';
+import React, { useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { useGetProductDetail } from '@/api/hooks/useGetProductDetail';
+import { Spinner } from '@/components/common/Spinner';
+import { ProductDetailContent } from '@/components/features/ProductDetail/ProductDetail/ProductDetailContent';
+import { useAuth } from '@/provider/Auth';
+
+const ProductDetail: React.FC = () => {
+  const { productId } = useParams<{ productId: string }>();
+  const { data, isLoading, isError } = useGetProductDetail(productId || '');
+  const navigate = useNavigate();
+  const isAuthenticated = useAuth();
+
+  useEffect(() => {
+    if (!isLoading && !data) {
+      navigate('/');
+    }
+  }, [productId, data, isLoading, navigate]);
+
+  const handleGiftButtonClick = () => {
+    if (!isAuthenticated) {
+      alert('로그인이 필요한 메뉴입니다.\n로그인 페이지로 이동하시겠습니까?');
+      navigate('/login');
+    } else {
+      if (data) {
```

Comment:
depth를 2개 가질 필요가 없어 보이네요
```suggestion
    } else if (data) {
```

Reply:
넵 수정하겠습니다.

Reply:
넵 !